### ISSUE TITLE
Proposal: Disable SA1649 for file names that contain a period

### ISSUE THREAD

**Issue by DavidDeSloovere** [DotNetAnalyzers/StyleCopAnalyzers#1710]
Given a class "ClassName" and a filename of "ClassName.Generated.cs", I think this should be valid for the rule. I believe it's a common practice for naming code generated files.

**Comment by sharwell** [comment]
It's also common practice for classes defined in generated files to be marked as a partial class, and file names are not checked for partial classes. For now, I recommend either marking the class as partial, or adding an `<auto-generated/>` file header to indicate that the file is generated. Aside from that, this is probably a duplicate of one or more of the following issues:

* #1430
* #1648 
* #1699

**Comment by DavidDeSloovere** [comment]
I'll look into the `partial`, although I'm not a fan of changing code just to pass a check that should fire anyway. So the `<auto-generated/>` might be the best solution for me.

Not really a dupe for those you list. I actually did look in the open issues for SA1649  ;-)

**Comment by sharwell** [comment]
@DavidDeSloovere The idea of ignoring SA1649 files with a `.` originally came up in #1662 (in a comment from @vweijsters).

I'm going to reopen this for discussion *specifically* on the topic of disabling SA1649 for files which contain a `.` in the name.

**Comment by sharwell** [comment]
:question: @vweijsters @oatkins What does StyleCop Classic do when you place a non-`partial` class `X` in a file named `X.Y.cs`? For example:

```csharp
// File X.Y.cs

public class X
{
}
```

**Comment by sharwell** [comment]
:memo: If StyleCop Classic ignored names of files of this form, I would be in favor of carrying over (and documenting) the additional exception when the `stylecop` file naming convention is used. If StyleCop Classic did not already make this exception, I would prefer simply relying on other exclusion mechanisms (adding `partial` or using `<auto-generated/>`) to cover the case.

**Comment by DavidDeSloovere** [comment]
Also wanted to add I already have a `[GeneratedCode]` attribute on my class. Might also be worth looking into skipping files with this attribute. [GeneratedCodeAttribute Class on msdn](https://msdn.microsoft.com/en-us/library/system.codedom.compiler.generatedcodeattribute(v=vs.110).aspx)

**Comment by sharwell** [comment]
> Also wanted to add I already have a `[GeneratedCode]` attribute on my class.

@DavidDeSloovere See #1094 for that. :smile:

**Comment by vweijsters** [comment]
@sharwell for your example StyleCop Classic reports SA1649

**Comment by berkeleybross** [comment]
:+1:  This would fix issue #1648

**Comment by sharwell** [comment]
@vweijsters Thanks, sounds like our current behavior already matches the old version.
